src/librustc_typeck/astconv.rs

```diff
@@ -1700,7 +1700,7 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx> + 'o {
                 // `Self` in impl (we know the concrete type).
                 assert_eq!(opt_self_ty, None);
                 self.prohibit_generics(&path.segments);
-                tcx.at(span).type_of(def_id)
+                self.normalize_ty(span, tcx.at(span).type_of(def_id))
```

Comment:
Add a comment so that we know we do this to normalize e.g. any unevaluated constants into their evaluated counterpart if possible.

Reply:
Would '// Try to evaluate any associated constants' be an accurate way of phrasing it?

Reply:
It's not really associated constants. Maybe

```
// Try to evaluate array length constants
```

?

Reply:
Thanks.